lib/datadog/appsec/processor.rb

```diff
@@ -39,6 +39,27 @@ def finalize
         end
       end
 
+      class InvalidContextError < StandardError; end
+
+      class << self
+        def current_context
+          Thread.current[:datadog_current_waf_context]
+        end
+
+        def current_context=(context)
+          unless context.instance_of?(Context)
+            raise InvalidContextError,
```

Comment:
A standard error like `TypeError` or `ArgumentError` is fine here. I'd rather not multiply exception types. WDYT?

Reply:
e.g https://github.com/DataDog/dd-trace-rb/blob/63b55b5a417b8d0e8580499759b5dece4971b432/lib/datadog/kit/identity.rb#L36